File: backend/internal/models/customer.go
package models

import "time"

type Customer struct {
	ID               uint      `gorm:"primaryKey" json:"id"`
	Name             string    `gorm:"size:200;not null" json:"name"`
	Phone            string    `gorm:"size:30" json:"phone"`
	Email            string    `gorm:"size:100" json:"email"`
	Address          string    `gorm:"size:500" json:"address"`
	Poin             int64     `gorm:"default:0" json:"poin"`
	TotalTransactions int64    `gorm:"default:0" json:"total_transactions"`
	TotalSpent       int64     `gorm:"default:0" json:"total_spent"`
	Notes            string    `gorm:"size:1000" json:"notes"`
	IsActive         bool      `gorm:"default:true" json:"is_active"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

File: tailadmin/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  phone: string;
  email: string;
  address: string;
  poin: number;
  total_transactions: number;
  total_spent: number;
  notes: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CustomerInput {
  name: string;
  phone?: string;
  email?: string;
  address?: string;
  notes?: string;
  is_active?: boolean;
}
